content/developer/reference/guidelines.rst

```diff
@@ -999,17 +999,88 @@ More precise JS guidelines are detailed in the `github wiki  <https://github.com
 You may also have a look at existing API in Javascript by looking Javascript
 References.
 
-CSS coding guidelines
----------------------
+Template guidelines
+-------------------
+
+Structure
+~~~~~~~~~
+
+.. note:: These rules are in place because the general structure of the template is considered more important than the details of each line.
+
+- Each block element should be on its own line.
+- Small inline/style/text elements can be placed either on the same line as their parent or on their own line (depending on their importance and size).
+- Indentation of 4 spaces is applied between a parent and its children elements. The closing tag of a parent should always be at the same indentation level as its opening tag.
+- It is preferable to use instructions (such as `t-if`) directly on the element on which they apply (as opposed to having them on a separate `<t>` element).
+
+Naming
+~~~~~~
+
+The root element and each other *important* element of the template should have a `name` attribute to ease the readability of the template.
+
+.. note::
+    The importance of an element in regard to this rule is left to the appreciation of the reviewer. If the structure of the template is hard to follow when reading just the first words of each line, it is probably a sign that some `name` attributes are missing on those lines.
+
+The name on the root element should follow the format `o-ComponentName`. And for children elements `o-ComponentName-partName` where:
+
+- `o-` is a must have Odoo prefix.
+- `ComponentName` is the PascalCase name of the current component/template.
+- `-` is a mandatory hyphen to separate the component name and the part name.
+- `partName` is the camelCase name of the child element referred by this `name` attribute.
+
+Attributes
+~~~~~~~~~~
+
+All the attributes of an element should be on the same line as the element itself. No character limit per line.
+
+.. note::
+    For extremely long lines, it is left to the appreciation of the reviewer to accept multi-line attributes within an element (but it is not mandatory).
+
+    In that case each attribute should be on its own line, with an indentation of 4 spaces relative to the element itself. The closing of the element (either `>` or `/>`) should be on its own line as well, with the same indentation as the corresponding opening character.
```

Comment:
ok for this. makes sense

Reply:
Are we talking about doing that?
```xml
<div class="hello"
     id="world"
>
    ...
</div>
```
I am not sure to interpret the sentence correctly

Reply:
Suggestion in that case would be
```xml
<div
    class="hello"
    id="world"
>
    ...
</div>
```
This is how it looks in a real example, according to the link @ged-odoo sent above: the https://github.com/odoo/odoo/blob/master/addons/web/static/src/webclient/navbar/navbar.xml#L26

I personally think both in this example and in all of that file it's way too verbose in terms of number of lines to apply this syntax.

I would be completely in favor of this syntax but only if the structure of the template was defined separately from its content. But when both are mixed like this and the template is more than a few lines, it becomes impossible to see the big picture of what the interface will contain, which I believe is more important very specific attributes of each element.

Note that as I said in another comment, this problem could actually become a non-problem with proper tooling. Tooling could go both ways too, either helping to expand for those that like it that way, or helping to collapse when the structure matters. Tooling could also be much more advanced, to define a template completely differently (for example with a drag&drop GUI like the editor, just to name an example) and then it would produce the expected result in whatever format we like to review/to have in the codebase, without even having to worry about it.

Reply:
the tooling exists: you can already do fold all/unfold all or fold at level N in most IDE. And the good thing is that it is an opt-in tooling, noone is forced to install it

Reply:
Agreed... most editors would prevent any need for these guidelines. I even have a XML parser in mine.
![Screenshot from 2021-07-07 10-35-12](https://user-images.githubusercontent.com/10338094/124727487-0e28f200-df0f-11eb-9302-ca3ad7cfa655.png)


Sorry but I physically cannot put a `>` on a single line, that just look like a buggy misplaced character *for me*. And maybe I don't read enough but I have never seen anyone doing that :shrug: 

> it becomes impossible to see the big picture of what the interface will contain, which I believe is more important than very specific attributes of each element.

I really wonder why you think that. If you want to see the "big picture", just fold level 1 of your XML. Once you have seen the big picture and start unfolding... many convention here will *prevent* you to see the specific attributes correctly. Like I don't mind the one-line attributes (I do it myself most of the time now)... but that just should not be a convention to enforce. Sometimes one-line is clearer, sometimes it's not, that should be left to case-by-case judging :shrug: Typically if I have a long and important conditional attribute, I might want to put it on its dedicated line for the sake of readability and git history.

Reply:
> This is how it looks in a real example, according to the link @ged-odoo sent above: https://github.com/odoo/odoo/blob/master/addons/web/static/src/webclient/navbar/navbar.xml#L26

If you check line 18, you will see yourself that it is not indented of 4 space, and it is probably a good thing ! 
As say by @qsm-odoo it should be left to case-by-case judging. 

```xml
<a id='ok' href='ok'>
```
```xml
<a t-if='a very long condition or another very long condition'
   id='ok'
   href='ok'>
```

```xml
<a t-if='a very long condition or another very long condition'
   id='ok'
   href='ok'
>
```
 
```xml
<input t-if='a very long condition or another very long condition'
    id='ok'
    href='ok'
>
```

```xml
<input t-if='a very long condition or another very long condition'
       id='ok'
       href='ok'
>
```

```xml
<input
    t-if='a very long condition or another very long condition'
    id='ok'
    href='ok'
>
```

```xml
<input
    t-if='a very long condition or another very long condition'
    id='ok' href='ok'
>
```

Imo none are wrong ! And sometimes mix both could make sense:
![image](https://user-images.githubusercontent.com/4848797/124731840-e63b8d80-df12-11eb-994f-a1982f04fb23.png)


 
For extremely long lines => Imo, when you want and when it make sense. And easy for review, history, and coding...
![image](https://user-images.githubusercontent.com/4848797/124731861-eb98d800-df12-11eb-92c4-a72aab959018.png)